### PR TITLE
[Storage] Resolve some help issue.

### PR DESCRIPTION
**Pull request by blueww**
## Description
Update Storage help to resolve some help issue:
https://github.com/Azure/azure-powershell/issues/5122
https://github.com/Azure/azure-powershell/issues/4979
https://github.com/Azure/azure-powershell/issues/5122

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Storage/Commands.Management.Storage/help/Update-AzureRmStorageAccountNetworkRuleSet.md [hunk 1]
```diff
@@ -149,7 +149,7 @@ Accept wildcard characters: False
 ```
 
 ### -VirtualNetworkRule
-The Array of VirtualNetworkRule objects to update to the NetworkRule Property of a Storage Account.
+The Array of VirtualNetworkRule objects to update to the NetworkRule Property of a Storage account.
```
Comment (markcowl) [inline src/ResourceManager/Storage/Commands.Management.Storage/help/Update-AzureRmStorageAccountNetworkRuleSet.md]: Please revert the changes in the .psm1 files below this

Comment (blueww) [reply src/ResourceManager/Storage/Commands.Management.Storage/help/Update-AzureRmStorageAccountNetworkRuleSet.md]: The *.psm1 files are removed.
How can I avoid these file generating in PowerShell developing?
Thanks!

**Comment by blueww** [comment]
@markcowl 

I have resolved your comments for this PR.
Would you please help to review and merge if no other issues?

And would you please merge the code in this branch back to preview branch when the PR in?
The code change on the branch are all based on public release package now.

File: src/ResourceManager/Storage/Commands.Management.Storage/help/Update-AzureRmStorageAccountNetworkRuleSet.md (hunk 1 above)
Comment (markcowl) [reply src/ResourceManager/Storage/Commands.Management.Storage/help/Update-AzureRmStorageAccountNetworkRuleSet.md]: When you run Update-MarkdownHelp, or Update-MarkdownHelpModule, you can just specify a single cmdlet, or your own module

File: src/ResourceManager/Storage/Commands.Management.Storage/help/Get-AzureRmStorageAccountNetworkRuleSet.md [hunk 2]
```diff
@@ -18,21 +18,21 @@ Get-AzureRmStorageAccountNetworkRuleSet [-ResourceGroupName] <String> [-Name] <S
...
 PS C:\> Get-AzureRmStorageAccountNetworkRuleSet  -ResourceGroupName "rg1" -AccountName "mystorageaccount"
 ```
 
-This command gets NetworkRule property of a specified storage account
+This command gets NetworkRule property of a specified Storage account
 
 ## PARAMETERS
 
 ### -DefaultProfile
-The credentials, account, tenant, and subscription used for communication with azure.
```
Comment (maddieclayton) [inline src/ResourceManager/Storage/Commands.Management.Storage/help/Get-AzureRmStorageAccountNetworkRuleSet.md]: Something happened here - DefaultProfile should not be changed here.  Can you regenerate this file?

File: src/ResourceManager/Storage/Commands.Management.Storage/help/New-AzureRmStorageAccount.md [hunk 3]
```diff
@@ -148,7 +148,7 @@ Accept wildcard characters: False
 ```
 
 ### -EnableHttpsTrafficOnly
-Indicates whether or not the Storage Account only enable https traffic.
+Indicates whether or not the Storage account only enables HTTPS traffic.”
```
Comment (maddieclayton) [inline src/ResourceManager/Storage/Commands.Management.Storage/help/New-AzureRmStorageAccount.md]: Take out the "

File: src/ResourceManager/Storage/Commands.Management.Storage/help/Set-AzureRmStorageAccount.md [hunk 4]
```diff
@@ -391,7 +391,7 @@ Accept wildcard characters: False
 ```
 
 ### -StorageEncryption
-Whether to set Storage Account Encryption KeySource to Microsoft.Storage or not.
+Whether or not to set the Storage account encryption to use Microsoft-managed keys.
```
Comment (maddieclayton) [inline src/ResourceManager/Storage/Commands.Management.Storage/help/Set-AzureRmStorageAccount.md]: Might be better as "Indicates whether or not to set the Storage account encryption to use Microsoft-managed keys."

File: src/Storage/Commands.Storage/help/Get-AzureStorageContainer.md [hunk 5]
```diff
@@ -86,6 +86,7 @@ Accept wildcard characters: False
 ### -Context
 Specifies the storage context.
 To create it, you can use the New-AzureStorageContext cmdlet.
+The cmdlet will fail when use a storage context created from SAS Token, since the cmdlet will get container Permissions which need Storage account key permission.
```
Comment (maddieclayton) [inline src/Storage/Commands.Storage/help/Get-AzureStorageContainer.md]: Should be "The cmdlet will fail when you use a storage context created from SAS Token because the cmdlet will query for container permissions which require Storage account key permission."

**Comment by blueww** [comment]
@maddieclayton 
I have fixed all you comments. And yes, we should target this PR to perview branch after following PR in on preview branch  https://github.com/Azure/azure-powershell/pull/5178.

For the description of "-DefaultProfile", I have revert them to lower case "azure" in this PR. But we should use upper case "Azure" instead of "azure" as it's branding. So would you like to modify the help generator to generate help with upper case "Azure"?

**Comment by maddieclayton** [comment]
I have made that change in [this PR](https://github.com/Azure/azure-powershell/pull/5191).  You can regenerate your help files after this PR is merged if you want it to be reflected in your PR.

**Comment by blueww** [comment]
Will target to preview after https://github.com/Azure/azure-powershell/pull/5190 is merged

**Comment by markcowl** [comment]
@blueww Now that AzureRM.StorageV2 branc is merged into preview, please rebase this on preview so that it doesn't re-make the changes from that PR.

**Comment by blueww** [comment]
@markcowl 
I have rebased this on preview.

**Comment by maddieclayton** [comment]
@blueww If you would like the -DefaultProfile change to be reflected in this PR, please regenerate the storage files so that this change is included.  Thanks!

**Comment by blueww** [comment]
@maddieclayton 
I have tried to regenerate the storage help file, but seems the -DefaultProfile is not updated.
I updated it manually in the PR.

**Comment by maddieclayton** [comment]
@blueww Please close the linked issues as fixed.

**Comment by blueww** [comment]
@maddieclayton 
All related issues closed, thanks!
